src/space.js

```diff
@@ -263,20 +268,46 @@ export default class ColorSpace {
 		return [...new Set(Object.values(ColorSpace.registry))];
 	}
 
+	static addToRegistry (id, space) {
+		if (this.registry[id] && this.registry[id] !== space) {
+			throw new Error(`Duplicate color space registration: '${id}'`);
+		}
+		this.registry[id] = space;
+	}
+
+	static validateIds () {
+		// Get full list of IDs.
+		const ids = this.all.flatMap(space =>
+			[space.id, ...(space.formats.color.ids ?? [])]
+		).sort();
+
+		let prev;
+		for (let id of ids) {
+			if (id === prev) {
+				throw new Error(`Duplicate color ID registration: '${id}'`);
+			}
+			prev = id;
+		}
+	}
+
 	static register (id, space) {
 		if (arguments.length === 1) {
 			space = arguments[0];
 			id = space.id;
 		}
 
 		space = this.get(space);
-		this.registry[id] = space;
-		return space;
-	}
+		this.addToRegistry(id, space);
 
-	static create (options) {
-		let space = new this(options);
-		return this.register(space);
+		if (space.aliases) {
+			for (let alias of space.aliases) {
+				this.addToRegistry(alias, space);
+			}
+		}
+
+		this.validateIds();
```

Comment:
I like detailed errors as much as the next person, but checking that deeply this seems a little too much.

Reply:
Do you mean we should forgo the ID check altogether? I think that's okay. As far as I can tell, if there is a collision, it'll simply use the first match, which is probably an acceptable behaviour?